### ISSUE TITLE
Clarify/fix online property

### ISSUE THREAD

**Issue by arv** [rocicorp/replicache-sdk-flutter#84]
https://github.com/rocicorp/replicache-sdk-flutter/blob/master/lib/src/replicache.dart#L82

It is not clear what the semantics of this property is.

`beginSync`, `endSync` and the replays can fail for different reasons. 

Does it meant that `pull` failed? `push`?

We need to clarify what the intent is and then make the code actually match that.

@aboodman

**Comment by aboodman** [comment]
Good point, sorry I should have documented this more clearly.

This is unfortunately one downside of separating push and pull into separate RPCs. It's cool that they can proceed separately but presents or a more complex model to developers and ultimately users.

Technically what _online represents is whether the last iteration of the sync protocol succeeded.

I think that probably what matters most to the developer and user is whether _pull_ is working. Ideally, I think, you'd tie your online status indicator to that.

**Comment by arv** [comment]
> Technically what _online represents is whether the last iteration of the sync protocol succeeded.

I don't think this is what the code does. We do not throw if push or pull fails. The information whether these succeeded or failed is in the `batchPushInfo` and the `clientViewInfo`.

**Comment by aboodman** [comment]
On Fri, May 22, 2020 at 6:38 PM Erik Arvidsson <notifications@github.com>
wrote:

> Technically what _online represents is whether the last iteration of the
> sync protocol succeeded.
>
> I don't think this is what the code does. We do not throw if push or pull
> fails. The information whether these succeeded or failed is in the
> batchPushInfo and the clientViewInfo.
>
It's all a bit of a mess and there's a bug on me to rationalize, but it's
definitely true that when I turn off the network connection (e.g., with
little snitch) an exception gets thrown and online gets flipped to false.
Maybe something else is going wrong though causing that by accident.

In any case what I describe is what I think it should do.
